cirq/contrib/acquaintance/permutation.py

```diff
@@ -143,12 +144,12 @@ def permutation(self) -> Dict[int, int]:
     def _decompose_(self, qubits: Sequence['cirq.Qid']) -> 'cirq.OP_TREE':
         yield self.swap_gate(*qubits)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return ('cirq.contrib.acquaintance.SwapPermutationGate(' +
                 ('' if self.swap_gate == ops.SWAP else repr(self.swap_gate)) +
                 ')')
 
-    def _value_equality_values_(self):
+    def _value_equality_values_(self) -> Any:
```

Comment:
Is Any the correct type here?  Should it be a Tuple or Iterable of Any instead?

Same with the others below.

Reply:
Yes. More precisely `Any` is the most general type supported by the [value equality](https://github.com/quantumlib/Cirq/blob/master/cirq/value/value_equality.py#L27) protocol. In practice, if you like you can narrow it down to a more concrete type (since mypy allows return type covariance), but this may cause problems in derived classes (if they want to use the more general type). Since we subclass `PermutationGate` quite a lot I decided to use the most general type allowed. We did the same in quirk's [arithmetic operation class](https://github.com/quantumlib/Cirq/blob/master/cirq/interop/quirk/cells/arithmetic_cells.py#L75).

Reply:
There is another reason `Any` is better here but not in `_circuit_diagram_info_`: we want values returned from `_value_equality_values_` to be treated as opaque.